compiler/nir/nirinsts.nim

```diff
@@ -108,6 +108,15 @@ type
     ObjConv,
     TestOf,
     Emit,
+
+    AsmGlobal # top-level basic asm. Better to use extended asm if it possible
+    Asm, # extended asm
+    AsmTemplate,
+    AsmInjectExpr, # like cexpr in gcc
+    AsmOutputOperand,
+    AsmInputOperand,
+    AsmClobber,
+
```

Comment:
Too GCC specific, should use `Emit` as the umbrella opcode. `Emit` having a list of `Verbatim` and `SymUse` etc opcodes. But it needs a previously missing `Verbatim` opcode that takes a string literal like `StrVal`.

Reply:
Probably also needs new pragmas that specify which section the Emit refers to.

Reply:
1. I understand correctly that asm stategments should be in Emit and it is universal for asembler and emitting code ? Then probably we should enter an enum with all uses and get it from Emit. For example:
```
Emit {
  EmitTarget "Asm" # or EmitTarget "EmitCode"
}.
```
If so, that's really great
2. What is Verbatim doing ? I hope not for this kind of code:
```
Emit {
  Verbatim "AsmTemplate"
  StrVal "
   mov %1, %0\n
   add $1, %0\n
  "
  Verbatim "OutputOperand"
  SymUse 3 #result var
  StrVal "" #no asm symbolic name
  StrVal "=r"
  ...
}.
```
This is tricky for code generation, perhaps this is better:
```
Emit {
  Verbatim "AsmTemplate" {
    StrVal "
      mov %1, %0
      add $1, %0\n
    "
  }
  Verbatim "OutputOperand" {
    SymUse 3 #result var
    StrVal "" #no asm symbolic name
    StrVal "=r"
  }
  ...
}
```
And maybe allowed verbatim values must be enum.